pkg/proxy/userspace/proxier.go

```diff
@@ -510,20 +577,41 @@ func (proxier *Proxier) unmergeService(service *v1.Service, existingPorts sets.S
 	}
 }
 
+func (proxier *Proxier) processServiceEvent(event int, oldService, service *v1.Service) {
+	e := serviceEvent{
+		event:      event,
+		service:    service.DeepCopy(),
+		oldService: oldService.DeepCopy(),
+	}
+
+	proxier.mu.Lock()
+	proxier.serviceEvents = append(proxier.serviceEvents, e)
```

Comment:
If there are a lot of events (enough to cause problems syncing), won't this balloon memory usage as you save copies of every Service and Endpoints snapshot?

Reply:
@thockin it only copies Services; but added a max queue depth that will trigger an immediate sync regardless of the ratelimiting.